### ISSUE TITLE
Enhancement: [no-unsafe-return] Add an option for ignoring the functions whose return type is specified

### ISSUE THREAD

**Issue by zanminkian** [typescript-eslint/typescript-eslint#10439]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unsafe-return/

### Description

Currently, this rule will report linting error on the code below:

```ts
interface Result {
  name: string
}

function parse(): Result {
  return JSON.parse('{"name":"foo"}');
}
```

But I think it's safe since the return type is specified.
So, adding an option to ignore the functions whose return type is specified is reasonable.

Option name: `allowReturnTypeSpecified`?

### Fail

When `allowReturnTypeSpecified` is false: 

```TypeScript

interface Result {
  name: string
}

function parse(): Result {
  return JSON.parse('{"name":"foo"}');
}
```

### Pass

When `allowReturnTypeSpecified` is true: 

```TypeScript

interface Result {
  name: string
}

function parse(): Result {
  return JSON.parse('{"name":"foo"}');
}
```

**Comment by bradzacher** [comment]
> But I think it's safe since the return type is specified.

This is the opposite of true

```ts
interface Result {
  name: string
}

function parse(): Result {
  return JSON.parse('1');
} 
```

With this change your code is now completely broken.

The return type being specified doesn't make anything safe. If anything it makes the whole situation even more *_unsafe_* because you're implicitly converting the `any` to a non-any; whereas if there was no annotation then the return type would be `any`.

As this option substantially weakens the rule without any DevX improvements or otherwise - we shall reject the option.
